### ISSUE TITLE
Newtonsoft.Json dependency decreased on NuGet latest packages (Graph 1.5.0 and Graph.Core 1.6.0)

### ISSUE THREAD

**Issue by nrobert** [microsoftgraph/msgraph-sdk-dotnet#171]
There has been a change in NuGet dependency for Microsoft.Graph and Microsoft.Graph.Core packages: 
- Previous packages: for Graph 1.4.0 / Graph.Core 1.5.0: Newtonsoft.Json (>= 10.0.2)
- Latest package: for Graph 1.5.0 / Graph.Core 1.6.0: Newtonsoft.Json (>= 6.0.1 && <= 10.0.2)

As a consequence, latest Newtonsoft.Json package is not supported 10.0.3.
Is it normal/voluntary?

Thanks

**Comment by cbales** [comment]
We wanted to continue to allow users to use older versions of Json.NET, which is why we reinstated the ranged approach to this package. Unfortunately, this work was done right before the latest version of Json.NET was released. 10.0.3 will be supported in the next release, in September.

**Comment by nrobert** [comment]
Ok @cbales thanks for the feedback

**Comment by XristophD** [comment]
I don't understand why it is not possible to add ">= 6.0.1" as a constraint to the dependencies and get rid of the limitation for newer versions? 

Now, you need to create a new package everytime Newtonsoft adds a new version - are there any breaking changes in 10.0.3 you need to be aware of?

**Comment by MIchaelMainer** [comment]
@XristophD While it is possible for us to set the dependency to >=6.0.1, we've decided that we want to confirm that there are no breaking changes when Newtonsoft.Json bumps a major version. We've got a pull request that changes our dependency spec to use [6.0.x, 11). We'll be updating the Nuget package soon. After version 11 comes out, we'll be updating our dependency on Newtonsoft after we test against it. The same will go for future versions.
